lib/new_post_manager.rb

```diff
@@ -92,7 +92,7 @@ def self.post_needs_approval?(manager)
       return :post_count
     end
 
-    return :trust_level if user.trust_level < SiteSetting.approve_unless_trust_level.to_i
+    return :group if user.groups.any? && (user.groups.pluck(:id) & self.allowed_groups).empty?
```

Comment:
We also have shorthand for this, do `user.in_any_groups?(SiteSetting.approve_unless_allowed_groups_map)`